synapse/storage/databases/main/events.py

```diff
@@ -2435,17 +2435,34 @@ def _update_backward_extremeties(
             "DELETE FROM event_backward_extremities"
             " WHERE event_id = ? AND room_id = ?"
         )
+        backward_extremity_tuples_to_remove = [
+            (ev.event_id, ev.room_id)
+            for ev in events
+            if not ev.internal_metadata.is_outlier()
+            # If we encountered an event with no prev_events, then we might
+            # as well remove it now because it won't ever have anything else
+            # to backfill from.
+            or len(ev.prev_event_ids()) == 0
+        ]
         txn.execute_batch(
             query,
-            [
-                (ev.event_id, ev.room_id)
-                for ev in events
-                if not ev.internal_metadata.is_outlier()
-                # If we encountered an event with no prev_events, then we might
-                # as well remove it now because it won't ever have anything else
-                # to backfill from.
-                or len(ev.prev_event_ids()) == 0
-            ],
+            backward_extremity_tuples_to_remove,
+        )
+
+        # Since we no longer need these backward extremities, it also means that
+        # they won't be backfilled from again so we no longer need to store the
+        # backfill attempts around it.
+        query = """
+            DELETE FROM event_backfill_attempts
+            WHERE event_id = ?
+        """
+        backward_extremity_event_ids_to_remove = [
+            (extremity_tuple[0],)
+            for extremity_tuple in backward_extremity_tuples_to_remove
```

Comment:
```suggestion
            (event_id,)
            for event_id, _room_id in backward_extremity_tuples_to_remove
```
.... that said, I think you may soon need to change this to include the room IDs :-)